### PR TITLE
Make ActiveShotBlock Optional

### LINKED ISSUE #57: Optional<ShotBlock> in return of activeShotBlock
The return type of `getActiveShotBlock` in `ControllerManager` should be changed to `Optional<ShotBlock>`. This removes unnecessary `nullPointerExceptions` in current and future code.

An optional is created via this method:

``` java
SoundCard soundcard = new Soundcard();
Optional<Soundcard> sc = Optional.of(soundcard);
```

### PR DESCRIPTION
**Pull request by geenen124**
Resolves #57

### CODE REVIEW

File: src/main/java/control/ControllerManager.java [hunk 1]
```diff
@@ -146,7 +148,7 @@ private void handleCancel(MouseEvent event) {
      *
      * @param block ShotBlock to set as active
      */
-    public void setActiveShotBlock(ShotBlock block) {
+    public void setActiveShotBlock(Optional<ShotBlock> block) {
```
Comment (martijnjanssen) [inline src/main/java/control/ControllerManager.java:151]: I would change the set to an actual `ShotBlock`, you can still put in a `null` variable if I am correct. Why would I want to set it to something if there is a possibility there is nothing in the variable I am setting it with?

Comment (martijnjanssen) [inline src/main/java/control/ControllerManager.java:151]: The `Optional.of(...)` action should be in this setter.

Comment (geenen124) [inline src/main/java/control/ControllerManager.java:151]: Won't work due to the fact that `Optional.of()` throws an exception when given null. And the point of using Optional would be to get rid of `null` variables anyways. So you could give it `Optional.empty()` instead

**Comment by Markvanderuit** [comment]
Looks fine to me. We can merge this once comments have been processed.

File: src/main/java/control/ControllerManager.java (hunk 1 above)
Comment (martijnjanssen) [inline src/main/java/control/ControllerManager.java:151]: The `null` check could very well be done inside the setter, allowing you to set it to `null`, resulting in an empty optional. Saves a lot of redundant code.

Comment (bart-de-jonge) [inline src/main/java/control/ControllerManager.java:151]: I agree with martijn's point, the setter should be a normal ShotBlock and you should do the null check in the setter

**Comment by bart-de-jonge** [comment]
I will close this PR until the team figured out what we want with this

**Pull request closed without merging**